controllers/space/SpaceController.php

```diff
@@ -2,44 +2,67 @@
 
 namespace humhub\modules\smartVillage\controllers\space;
 
+use Firebase\JWT\JWT;
+use humhub\modules\rest\models\ConfigureForm;
 use humhub\modules\smartVillage\components\AuthBaseController;
 use humhub\modules\space\models\Space;
 use humhub\modules\rest\definitions\SpaceDefinitions;
+use humhub\modules\user\models\User;
 
 class SpaceController extends AuthBaseController
 {
     public function actionFind(){
         $results = [];
-        $query = Space::find()->where(['visibility' => Space::VISIBILITY_ALL]);
+
+        //Check the requested user is guest or not
+        $user = $this->checkUserIsRegistered();
+        if($user == false){
+            $query = Space::find()->where(['visibility' => Space::VISIBILITY_ALL]);
+        }else{
+            $query = Space::find()->where(['visibility' => Space::VISIBILITY_REGISTERED_ONLY]);
+
+        }
 
         $pagination = $this->handlePagination($query);
         foreach ($query->all() as $space) {
             $results[] = SpaceDefinitions::getSpace($space);
         }
         if(count($results) > 0){
             return $this->returnPagination($query, $pagination, $results);
-        }else{
-            return $this->returnError(400,"No space data is publicly available!");
         }
     }
 
     public function actionView($spaceId){
-        $checkSpace = Space::findOne($spaceId);
-        $checkVisibility = Space::find()->where(['id'=> $spaceId])->andWhere(['visibility' => Space::VISIBILITY_ALL])->one();
+        $space = Space::findOne($spaceId);
 
-        if(!isset($checkSpace)){
+        if(!isset($space)){
             return $this->returnError(400,"Space not found!");
         }
 
-        if(!isset($checkVisibility)){
-            return $this->returnError(400,"Guest user cannot read this space data");
+        $user = $this->checkUserIsRegistered();
+        if($user == false){
+            if($space->visibility == Space::VISIBILITY_REGISTERED_ONLY){
```

Comment:
this should also apply for `VISIBILITY_NONE` or am i wrong? guest users should not be able to see private spaces

Reply:
Yes, you are right and add one more condition in this logic